src/io/iohk/decco/PartialCodec.scala

```diff
@@ -54,6 +54,25 @@ trait PartialCodec[T] { self =>
       self.encode(u2t(u), start, destination)
   }
 
+  final def mapOpt[U](t2u: T => Option[U], u2t: U => T): PartialCodec[U] = new PartialCodec[U] {
```

Comment:
Why `Option` and not `Either`?

Reply:
The failure has no information in it, so there is nowhere I could put faling information into. I need this for a very specific purpose and Option is enough. In the future this can/should be improved, but I don't think it's needed right now.